kratos/linear_solvers/fallback_linear_solver.h

```diff
@@ -0,0 +1,897 @@
+            }
+        }
+        return success;
+    }
+
+    /**
+     * @brief Eigenvalue and eigenvector solve method for derived eigensolvers
+     * @param K The stiffness matrix
+     * @param M The mass matrix
+     * @param Eigenvalues The vector containing the eigen values
+     * @param Eigenvectors The matrix containing the eigen vectors
+     */
+    void Solve(
+        SparseMatrixType& K,
+        SparseMatrixType& M,
+        DenseVectorType& Eigenvalues,
+        DenseMatrixType& Eigenvectors
+        ) override
+    {
+        KRATOS_ERROR << "This is designed for linear system only, not for eigen values calculation" << std::endl;
+    }
+
+    /**
+     * @brief Checks if additional physical data is needed by the solver.
+     * @details Some solvers may require a minimum degree of knowledge of the structure of the matrix. 
+     * For instance, when solving a mixed u-p problem, it is important to identify the row associated with v and p.
+     * Another example is the automatic prescription of rotation null-space for smoothed-aggregation solvers,
+     * which require knowledge of the spatial position of the nodes associated with a given degree of freedom (DOF).
+     * @return True if additional physical data is needed, false otherwise.
+     */
+    bool AdditionalPhysicalDataIsNeeded() override
+    {
+        return GetCurrentSolver()->AdditionalPhysicalDataIsNeeded();
+    }
+
+    /**
+     * @brief Provides additional physical data required by the solver.
+     * @details Some solvers may require a minimum degree of knowledge of the structure of the matrix.
+     * For example, when solving a mixed u-p problem, it is important to identify the row associated with v and p.
+     * Another example is the automatic prescription of rotation null-space for smoothed-aggregation solvers,
+     * which require knowledge of the spatial position of the nodes associated with a given degree of freedom (DOF).
+     * This function provides the opportunity to provide such data if needed.
+     * @param rA The sparse matrix.
+     * @param rX The solution vector.
+     * @param rB The right-hand side vector.
+     * @param rDoFSet The set of degrees of freedom.
+     * @param rModelPart The model part.
+     */
+    void ProvideAdditionalData(
+        SparseMatrixType& rA,
+        VectorType& rX,
+        VectorType& rB,
+        typename ModelPart::DofsArrayType& rDoFSet,
+        ModelPart& rModelPart
+        ) override
+    {
+        GetCurrentSolver()->ProvideAdditionalData(rA, rX, rB, rDoFSet, rModelPart);
+    }
+
+    /**
+     * @brief Add a linear solver to the collection.
+     * @details This function adds a linear solver to the collection without extending parameters.
+     * @param pSolver Pointer to the linear solver to be added.
+     */
+    void AddSolver(LinearSolverPointer pSolver)
+    {
+        // Increase the solvers vector
+        mSolvers.push_back(pSolver);
```

Comment:
again, the internal `Parameters` don't get updated

Reply:
Parameters are not mean to be updated, and it is not trivial to deduce the aprameters from a given constructed linear solver.

Reply:
Fixed